ui/raidboss/data/06-ew/ultimate/dragonsongs_reprise_ultimate.ts

```diff
@@ -818,14 +827,306 @@ const triggerSet: TriggerSet<Data> = {
+              data.diveFromGracePreviousPosition[nameB] = 'east';
+            } else {
+              data.diveFromGracePreviousPosition[nameB] = 'west';
+              data.diveFromGracePreviousPosition[nameA] = 'east';
+            }
+          } else if (distanceAC === sorted[0]) {
+            data.diveFromGracePreviousPosition[nameB] = 'middle';
+            if (posAX < posCX) {
+              data.diveFromGracePreviousPosition[nameA] = 'west';
+              data.diveFromGracePreviousPosition[nameC] = 'east';
+            } else {
+              data.diveFromGracePreviousPosition[nameC] = 'west';
+              data.diveFromGracePreviousPosition[nameA] = 'east';
+            }
+          } else if (distanceBC === sorted[0]) {
+            data.diveFromGracePreviousPosition[nameA] = 'middle';
+            if (posAX < posCX) {
+              data.diveFromGracePreviousPosition[nameB] = 'west';
+              data.diveFromGracePreviousPosition[nameC] = 'east';
+            } else {
+              data.diveFromGracePreviousPosition[nameC] = 'west';
+              data.diveFromGracePreviousPosition[nameB] = 'east';
+            }
+          }
+        } else {
+          // Only comparing X values for Dive 2
+          if (posAX < posBX) {
+            data.diveFromGracePreviousPosition[nameA] = 'west';
+            data.diveFromGracePreviousPosition[nameB] = 'east';
+          } else {
+            data.diveFromGracePreviousPosition[nameB] = 'west';
+            data.diveFromGracePreviousPosition[nameA] = 'east';
+          }
+        }
+
+        // First Dive, on num1s
+        if (data.diveFromGraceTowerCounter === 1) {
+          // Stack or Move
+          if (num === 1) {
+            // Stack => Predict Tower 3
+            if (data.diveFromGracePreviousPosition[data.me] === 'middle')
+              return output.stackNorth!();
+            // Other players just move so 3s can get tower
+            return output.move!();
+          }
+        }
+
+        // Second Dive, on num2s
+        if (data.diveFromGraceTowerCounter === 2) {
+          // Stack => Predict Tower 3 (based on previous position)
+          if (num === 2) {
+            if (data.diveFromGracePreviousPosition[data.me] === 'west')
+              return output.spineshatterTower3!();
+            if (data.diveFromGracePreviousPosition[data.me] === 'east')
+              return output.elusiveTower3!();
+          }
+
+          // Call Tower 2 Soak (based on previous position)
+          if (num === 1) {
+            if (data.diveFromGracePreviousPosition[data.me] === 'west')
+              return output.spineshatterTower2!();
+            if (data.diveFromGracePreviousPosition[data.me] === 'east')
+              return output.elusiveTower2!();
+          }
+        }
+        // Third Dive, on num3s
+        if (data.diveFromGraceTowerCounter === 3) {
+          // Num3s need to move out of tower 3
+          if (num === 3)
+            return output.move!();
```

Comment:
Everybody has to move out of their tower (and are usually doing it as a part of the out/in).  Personally, I think we could skip this call.